src/Cassandra.Tests/TaskTests.cs

```diff
@@ -144,6 +144,8 @@ public void TaskHelper_TaskCompletionSourceWithTimeout_Does_Not_Invoke_Delegate_
             Assert.False(called);
         }
 
+        [Ignore("this test is not reliable at all, ReSharper shows there are " +
```

Comment:
I didn't realize the test was just looking for the production code while the analysis was going through all the code including the tests. Despite that, the original reason for this `Ignore` was that the test was failing in two parts:

```
using (cancellationToken.Register(() => cancellationCompletionSource.TrySetResult(true)))
            {
                if (task != await Task.WhenAny(task, cancellationCompletionSource.Task).ConfigureAwait(false))
                {
                    throw new OperationCanceledException(cancellationToken);
                }
            }
```

```
 /// <summary>
        /// Designed to await tasks with a cancellation token when the method that returns the task doesn't
        /// accept a token.
        /// </summary>
        /// <param name="task"></param>
```

I originally tried to find a regex that would not result in false positives for these two cases and I didn't find one. After doing the analysis I thought there were also false negatives so I decided to `Ignore` it. 

Now that I see that this is not the case I've changed the regex to fix the first false positive (the regex always expected a `;`) and changed the text of the comment from `await` to `Await` so that the regex doesn't pick it up.

What do you think about the new regex?

Reply:
Nice! using negative lookahead is more solid that lookbehind on the last portion of the statement 👍